core/src/main/java/com/adobe/aem/commons/assetshare/content/renditions/AssetRenditions.java

```diff
@@ -0,0 +1,141 @@
+ * limitations under the License.
+ *
+ */
+
+package com.adobe.aem.commons.assetshare.content.renditions;
+
+import com.adobe.aem.commons.assetshare.content.AssetModel;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.ValueMap;
+import org.apache.sling.api.wrappers.ValueMapDecorator;
+import org.osgi.annotation.versioning.ProviderType;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ *
+ */
+@ProviderType
+public interface AssetRenditions {
+    String VAR_ASSET_PATH = "${asset.path}";
+    String VAR_ASSET_NAME = "${asset.name}";
+    String VAR_ASSET_EXTENSION = "${asset.extension}";
+    String VAR_RENDITION_NAME = "${rendition.name}";
+
+    /**
+     * @return a list of all registered AssetRenditionDispatchers in the system ordered in Descending order by service.ranking.
+     */
+    List<AssetRenditionDispatcher> getAssetRenditionDispatchers();
+
+    /**
+     * Derives the renditionName from the request which is typically used to resolve the AssetRenditionDispatcher and the mapping there-in.
+     * <p>
+     * This typically pulls the first segment of the suffix, however this logic may change overtime if the scheme changes.
+     *
+     * @param request the request TO the AssetRenditionServlet.
+     *
+     * @return the renditionName parsed from the
+     */
+    String getRenditionName(SlingHttpServletRequest request);
+
+    /**
+     * Creates a URL to the AssetRenditionServlet for the
+     *
+     * @param request the request object.
+     * @param asset   the asset to rendition.
+     * @param params  the params which describe how the AssetRenditionServlet should handle sending the asset rendition to the response.
+     *
+     * @return the URL that can be used to HTTP GET Request the specified asset rendition (does NOT include scheme/host/port).
+     */
+    String getUrl(SlingHttpServletRequest request, AssetModel asset, UrlParams params);
+
+    /**
+     * Creates a Map that is used to provide the label/values to drive the AssetRenditionsDatasource dropdown.
+     * <p>
+     * This is a convenience method, however each AssetRenditionDispatcher implementation can implement their own getOptions() rather than wrapping a call to this method.
+     *
+     * @param mappings the raw mappings from the
+     *
+     * @return the Options is the format LABEL:VALUE
+     */
+    Map<String, String> getOptions(Map<String, ? extends Object> mappings);
+
+    /**
+     * Replaces the 'variables' in the expressions with the corresponding bits derived from the request.
+     *
+     * @param request the request object to the AssetRenditionServlet.
+     * @param expression the expression to replace the variables in.
+     * @return the expression with the variables replaced with values derived from the request.
+     */
+    String evaluateExpression(SlingHttpServletRequest request, String expression);
```

Comment:
Nope - this is a @ProviderType interface indicating the expectation is this bundle will only implement it.

If we wanted to allow ppl to provide their own expression evaluation it would be better to create a discrete interface for that use case, and make it as ConsumerType - but i think thats likely YAGNI - but also something we can add later if WGNI (We gunna need it)

https://osgi.org/javadoc/r6/annotation/org/osgi/annotation/versioning/ProviderType.html

Reply:
i think i missed the point - you were suggesting adding the default impl into the interface as the default function impl? Ill probably just leave it in the Impl file -- since i supposed you could make the argument all "base" impls could be moved into the default method on the interface which seems weird/confusing.